ui/src/main.ts

```diff
@@ -1,12 +1,27 @@
-import {enableProdMode} from '@angular/core';
+import {enableProdMode, ReflectiveInjector} from '@angular/core';
 import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
 
 import {AppModule} from './app/app.module';
 import {environment} from './environments/environment';
+import {EnvironmentConfigurationLoaderService} from "./environments/environment-configuration-loader.service";
 
 if (environment.production) {
   enableProdMode();
 }
 
-platformBrowserDynamic().bootstrapModule(AppModule)
-  .catch(err => console.log(err));
+let environmentConfigurationLoader = ReflectiveInjector
+  .resolveAndCreate([EnvironmentConfigurationLoaderService])
```

Comment:
Do you need to use a ReflectiveInjector here? Why can't you just do `new EnvironmentConfigurationLoaderService()`?

Reply:
From the Angular docs, it suggests to avoid using `new` for services as much as possible. Although `ReflectiveInjector` does provide advanced features in this case, it resolves dependencies between services when it's getting more complicated, based on my reading on the docs. So I personally prefer to avoid `new` here. 

However, as I did more research on this now, I should have used `StaticInjector` here because `ReflectiveInjector` is deprecated! Thanks for cathing this.